tests/integration/events.go

```diff
@@ -349,6 +351,40 @@ func getEventsForCreateDoc(s *state, action CreateDoc) map[string]struct{} {
 	return expect
 }
 
+func waitForKeyRetrievedEvent(s *state, nodeIDs []int) {
+	for nodeID := 0; nodeID < len(s.nodes); nodeID++ {
+		if len(nodeIDs) > 0 && !slices.Contains(nodeIDs, nodeID) {
+			continue
+		}
+
+		select {
+		case _, ok := <-s.nodeEvents[nodeID].encKeysRetrieved.Message():
+			if !ok {
+				require.Fail(s.t, "subscription closed waiting for key retrieved event")
+			}
+
+		case <-time.After(eventTimeout):
+			require.Fail(s.t, "timeout waiting for key retrieved event")
+		}
+	}
+}
+
+func waitForSync(s *state, action WaitForSync) {
+	count := int(action.Count)
```

Comment:
todo: This looks like it is wrong, and would result on serial waits and timeouts for concurrent operations.  In particular, this will make some failures very slow - e.g. if `count == 10`, then the effective timeout would be `10 * eventTimeout`.

Please rework this, I suggest using a go routine + waitGroup.

Reply:
done